### ISSUE TITLE
Malformed function and Wrong argument, sequencep, 1

### ISSUE THREAD

**Issue by fxleblanc** [flycheck/flycheck#908]
As explained in this issue, https://github.com/lifeisfoo/emacs-grails/issues/8, the following snippet:

```
((groovy-mode (grails . 1))
 (html-mode (grails . 1))
 (java-mode (grails . 1)))

;; To have grails mode always enabled uncomment this line
((nil . ((grails-mode . 1))))
```

in the file `.dir-locals.el` produces two errors in flycheck:
- `(groovy-mode (grails . 1))' is a malformed function
- Wrong type argument: sequencep, 1

However, the plugin still works.

**Comment by cpitclaudel** [comment]
That makes sense: we're running an ELisp checker on a file that does not contain ELisp code (but instead an unquoted version of a list data structure).

**Comment by cpitclaudel** [comment]
Should we just disable flycheck in .dir-locals files?

**Comment by lunaryorn** [comment]
@cpitclaudel Probably.  I always enjoyed Flycheck for syntax checking there, and ignored the false positives, but I can see that these are very confusing, and strictly speaking `.dir-locals.el` is not proper Emacs Lisp code, but rather just data.

**Comment by fxleblanc** [comment]
Is there a blacklist of filenames that i could set in my init.el? That could give more flexibility.

**Comment by lunaryorn** [comment]
@fxleblanc There's no direct option for this, and we will not add one for what is a small edge case in a single syntax checker.  However, you can easily write your own `emacs-lisp-mode-hook` to turn of Flycheck in `.dir-locals.el`.

**Comment by fxleblanc** [comment]
You're right. I'll try to figure something out.

**Comment by lunaryorn** [comment]
I'm closing this issue.  I don't think that there's anything for us to improve here.

**Comment by carrete** [comment]
> I don't think that there's anything for us to improve here.

**Comment by cpitclaudel** [comment]
@carrete Do you have a concrete idea of how we could improve this? One thing that we could do is ask Emacs to add a special mode for .dir-locals.el; then Flycheck wouldn't fire there.

**Comment by carrete** [comment]
> Do you have a concrete idea of how we could improve this?

I can't comment how this should be implemented. From the perspective of a user, I would be OK if the errors are eliminated or still present but with some additional annotation somewhere that says they should be ignored.

**Comment by cpitclaudel** [comment]
Right, my question is what circumstances you want the errors eliminated in.  Is it specifically in .dir-locals.el files, or are there other cases?

**Comment by carrete** [comment]
> Is it specifically in .dir-locals.el files, or are there other cases?

Gotcha. Sorry I didn't understand the question. I've only encountered this issue with .dir-locals.el. Thanks.

**Comment by cpitclaudel** [comment]
Thanks for your patience. Could you try the branch at https://github.com/flycheck/flycheck/tree/cpitclaudel_908 ?

**Comment by cpitclaudel** [comment]
I was hoping that we wouldn't have to fix this explicitly by instead adding a new mode in Emacs itself specific to dir-locals files, but this idea was rejected, so I've created a patch.
